src/main/scala/nl.knaw.dans.easy.bagstore/component/BagProcessingComponent.scala

```diff
@@ -175,22 +177,22 @@ trait BagProcessingComponent extends DebugEnhancedLogging {
       }
     }
 
-    def getReferenceBags(bagDir: Path): Try[Option[Path]] = Try {
+    def getReferenceBags(bagDir: Path, bagId: BagId): Try[Option[Path]] = {
       trace(bagDir)
       val refbags = bagDir.resolve("refbags.txt")
       if (Files.exists(refbags)) {
         // copy to tempDir
-        val tempRefbags = Files.createTempFile(stagingBaseDir, "refbags-", "")
-        Files.deleteIfExists(tempRefbags)
-        Files.move(refbags, tempRefbags)
-        assert(!Files.exists(refbags), s"$refbags should have been moved to $tempRefbags, however, it appears to still be present here")
-
-        // remove refbags.txt from all tagmanifests (if it was present there)
-        bagFacade.removeFromTagManifests(bagDir, "refbags.txt")
-
-        Some(tempRefbags)
+        for {
+          _ <- assertRefbagsFileIsNotEmpty(bagId, refbags)
+          tempRefbags = Files.createTempFile(stagingBaseDir, "refbags-", "")
+          _ = Files.deleteIfExists(tempRefbags)
+          _ = Files.move(refbags, tempRefbags)
+          _ = assert(!Files.exists(refbags), s"$refbags should have been moved to $tempRefbags, however, it appears to still be present here")
+          _ <- // remove refbags.txt from all tagmanifests (if it was present there)
```

Comment:
could you move this comment above the `_ <- ...` rather than as part of it?

```scala
// remove refbags.txt from all tagmanifests (if it was present there)
_ <- bagFacade.removeFromTagManifests(bagDir, "refbags.txt")
```

Reply:
i did not notice that, good catch